src/vmspawn/vmspawn.c

```diff
@@ -697,18 +798,28 @@ static int run_virtual_machine(int kvm_device_fd, int vhost_device_fd) {
                 log_warning("Couldn't find OVMF firmware blob with Secure Boot support, "
                             "falling back to OVMF firmware blobs without Secure Boot support.");
 
+        shm = arg_directory ? ",memory-backend=mem" : "";
+        if (ARCHITECTURE_SUPPORTS_SMM)
+                machine = strjoin("type=" QEMU_MACHINE_TYPE ",smm=", on_off(ovmf_config->supports_sb), shm);
+        else
+                machine = strjoin("type=" QEMU_MACHINE_TYPE, shm);
+        if (!machine)
+                return log_oom();
+
+        if (arg_linux) {
+                kernel = strdup(arg_linux);
+                if (!kernel)
+                        return log_oom();
+        } else if (arg_directory)
+                return log_error_errno(SYNTHETIC_ERRNO(ENOENT), "Please specify a kernel (--linux=) when using -D/--directory, refusing.");
```

Comment:
nitpick: we generally try to suffix switch that take parameters with a `=` in docs and log messages. Hence `--directory` → `--directory=` please